app/models/contribution.rb

```diff
@@ -79,6 +81,7 @@ class Contribution
     event :wipe do # named :wipe and not :delete because Mongoid::Document brings #delete
       transitions from: :draft, to: :deleted
```

Comment:
I think that we are introducing an ambiguity here that has potential to cause confusion: contributions that have never been published will not be wiped and moved to deleted status, but will be destroyed.

I suspect that we should introduce an [AASM guard](https://github.com/aasm/aasm#guards) to reflect this:
```ruby
       transitions from: :draft, to: :deleted do
         guard do
           ever_published?
         end
       end
```

We may then want to add method(s) reflecting whether a contribution may be either wiped or destroyed like:
```ruby
def destroyable?
  !ever_published?
end

def removable?
  wipeable? || destroyable?
end
```